specutils/tests/test_template_comparison.py

```diff
@@ -11,6 +11,66 @@
 # TODO: Add some tests that are outliers: where the observed and template do not overlap (what happens?),
 # TODO: where there is minimal overlap (1 point)?
 
+def test_template_match_no_overlap():
+    """
+    Test template_match when both observed and template spectra have no overlap on the wavelength axis
+    """
+    print("no overlap test")
+    # Seed np.random so that results are consistent
+    np.random.seed(42)
+
+    # Create test spectra
+    spec_axis = np.linspace(0, 50, 50) * u.AA
+    spec_axis_no_overlap = np.linspace(51, 102, 50) * u.AA
+
+    spec = Spectrum1D(spectral_axis=spec_axis,
+                      flux=np.random.randn(50) * u.Jy,
+                      uncertainty=StdDevUncertainty(np.random.sample(50), unit='Jy'))
+
+    spec1 = Spectrum1D(spectral_axis=spec_axis_no_overlap,
+                       flux=np.random.randn(50) * u.Jy,
+                       uncertainty=StdDevUncertainty(np.random.sample(50), unit='Jy'))
+
+    # Get result from template_match
+    tm_result = template_comparison.template_match(spec, spec1)
+
+    # Create new spectrum for comparison
+    spec_result = Spectrum1D(spectral_axis=spec_axis,
+                             flux=spec1.flux * template_comparison._normalize_for_template_matching(spec, spec1))
+
+    # assert quantity_allclose(tm_result[0].flux, spec_result.flux, atol=0.01*u.Jy)
+    assert np.isnan(tm_result[1])
+
+def test_template_match_minimal_overlap():
+    """
+    Test template_match when both observed and template spectra have minimal overlap on the wavelength axis
+    """
+    print("minimal overlap test")
+    # Seed np.random so that results are consistent
+    np.random.seed(42)
+
+    # Create test spectra
+    spec_axis = np.linspace(0, 50, 50) * u.AA
+    spec_axis_no_overlap = np.linspace(45, 95, 50) * u.AA
+
+    spec = Spectrum1D(spectral_axis=spec_axis,
+                      flux=np.random.randn(50) * u.Jy,
+                      uncertainty=StdDevUncertainty(np.random.sample(50), unit='Jy'))
+
+    spec1 = Spectrum1D(spectral_axis=spec_axis_no_overlap,
+                       flux=np.random.randn(50) * u.Jy,
+                       uncertainty=StdDevUncertainty(np.random.sample(50), unit='Jy'))
+
+    # Get result from template_match
+    tm_result = template_comparison.template_match(spec, spec1)
+
+    # Create new spectrum for comparison
+    spec_result = Spectrum1D(spectral_axis=spec_axis,
+                             flux=spec1.flux * template_comparison._normalize_for_template_matching(spec, spec1))
+
+    # assert quantity_allclose(tm_result[0].flux, spec_result.flux, atol=0.01*u.Jy)
+    assert np.isnan(tm_result[1])
+
```

Comment:
Should there be another `assert` to ensure that the places where they do overlap have non-`NaN` values?

Reply:
I'm checking now and it looks like with one element of overlap, there is still a full array of `NaN` values.